src/app/data-model-provider/MetadataTypes.h

```diff
@@ -108,10 +202,57 @@ struct AcceptedCommandEntry
 {
     CommandId commandId;
 
-    // TODO: this can be more compact (use some flags for privilege)
-    //       to make this compact, add a compact enum and make flags/invokePrivilege getters (to still be type safe)
-    BitFlags<CommandQualityFlags> flags;
-    Access::Privilege invokePrivilege = Access::Privilege::kOperate;
+    // Constructor
+
+    StartBitFieldInit // Disabling '-Wconversion' & '-Wconversion'
+
+        constexpr AcceptedCommandEntry(
+            CommandId id = 0,                                                                  // commandId initial value,
+                                                                                               // this could be altered later
+            CommandQualityFlags cmdQualityFlags         = static_cast<CommandQualityFlags>(0), // mask.flags initial value
+            std::optional<Access::Privilege> invokePriv = Access::Privilege::kView             // mask.invokePrivilege initial value
+            ) :
+        commandId(id),
+        mask{ to_underlying(cmdQualityFlags) & ((1 << 3) - 1), // Narrowing expression to 3 bits
+              to_underlying(invokePriv.value_or(static_cast<Access::Privilege>(0))) & ((1 << 5) - 1) }
+    // Narrowing expression to 5 bits
+    {}
+
+    EndBitFieldInit // Enabling '-Wconversion' & '-Wconversion'
+
+        // Getter for mask.invokePrivilege
+        constexpr std::optional<Access::Privilege>
+        GetInvokePrivilege() const
+    {
+        if (mask.invokePrivilege == 0)
+        {
+            return std::nullopt;
+        }
+        else
+        {
+            return static_cast<Access::Privilege>(mask.invokePrivilege);
+        }
```

Comment:
```suggestion
        return static_cast<Access::Privilege>(mask.invokePrivilege);
```

General readability rule: do not use `else` after return. Please apply throughout.